dashboard/app/models/user.rb

```diff
@@ -190,6 +191,9 @@ class User < ActiveRecord::Base
   has_many :districts_users, class_name: 'DistrictsUsers'
   has_many :districts, through: :districts_users
 
+  has_many :authentication_options, dependent: :destroy
+  belongs_to :primary_email, class_name: 'AuthenticationOption', foreign_key: :primary_email_id
```

Comment:
This seems oddly named. The full set is `authentication_options`, yet the primary one is `primary_email`? I'd expect `primary_email` to be an email. 

How about?
```ruby
belongs_to :primary_authentication_option, class_name: 'AuthenticationOption', foreign_key: :primary_authentication_option_id

def primary_email
  primary_authentication_option.try(:email)
end
```

Reply:
:+1: